### PR TITLE
Checkstyle - Javadoc - enforce formating for param, return and throw literals usage

### PR DESCRIPTION
**Pull request by vhvb1989**
image

### CODE REVIEW

File: sdk/core/azure-core-management/src/test/java/com/azure/core/management/implementation/Value.java [hunk 1]
```diff
@@ -6,7 +6,7 @@
 /**
  * A container for a generic type. Serves a similar purpose as pointers in C/C++. It's a workaround
  * for the fact that Java doesn't allow mutation of local variables in closure.
- * @param <T>
+ * @param <T> a generic type
```
Comment (JonathanGiles) [inline sdk/core/azure-core-management/src/test/java/com/azure/core/management/implementation/Value.java]: These kind of statements need improving - it should say something more useful like "The type of the object stored within the Value instance". Please do a pass over the PR to improve the text in these cases.

File: sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubAsyncProducerTest.java [hunk 2]
```diff
@@ -62,7 +62,7 @@ public void teardown() {
     }
 
     /**
-     * Verifies that sending multiple events will result in calling producer.send(List<Message>).
+     * Verifies that sending multiple events will result in calling {@code producer.send(List<Message>)}.
```
Comment (JonathanGiles) [inline sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubAsyncProducerTest.java]: This is a test. You don't need to worry about fixing up javadoc in test classes (add a suppression if necessary). We care about JavaDoc only in public API and samples.

File: sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/KeyClient.java [hunk 3]
```diff
@@ -501,6 +501,7 @@ public DeletedKey deleteKey(String name) {
      * <p><strong>Code Samples</strong></p>
      * <p>Gets the deleted key from the key vault enabled for soft-delete. Prints out the details of the deleted key
      * returned in the response.</p>
+     * <pre>
      * //Assuming key is deleted on a soft-delete enabled key vault.
      * {@codesnippet com.azure.keyvault.keys.keyclient.getDeletedKey#string}
      * </pre>
```
Comment (JonathanGiles) [inline sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/KeyClient.java]: The opening and closing pre statements here are redundant (shortly Shawn's CheckStyle should flag these). Please remove.

File: sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/AsymmetricEncryptionAlgorithm.java [hunk 4]
```diff
@@ -29,10 +29,10 @@ protected AsymmetricEncryptionAlgorithm(String name) {
...
      *
      * @param keyPair The key pair to use.
-     * @return
-     * @throws InvalidKeyException
-     * @throws NoSuchAlgorithmException
-     * @throws NoSuchPaddingException
+     * @return abstract {@link ICryptoTransform}
+     * @throws InvalidKeyException when key is not valid
+     * @throws NoSuchAlgorithmException if algorithm is not found
+     * @throws NoSuchPaddingException if padding is set wrong
```
Comment (JonathanGiles) [inline sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/AsymmetricEncryptionAlgorithm.java]: This exception text should be improved after this is merged (by @g2vinay)

File: sdk/keyvault/azure-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/SecretAsyncClient.java [hunk 5]
```diff
@@ -408,6 +408,7 @@
      * <pre>
      * //Assuming secret is deleted on a soft-delete enabled vault.
      * {@codesnippet com.azure.keyvault.secrets.secretclient.getDeletedSecret#string}
+     * </pre>
```
Comment (JonathanGiles) [inline sdk/keyvault/azure-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/SecretAsyncClient.java]: Same comments through here with pre tags being unnecessary and should be removed now.

File: eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/JavaDocFormatting.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+ *  Description text should only have one space character after the parameter name or '@return' statement.
+ *  Text should not start on a new line or have any additional spacing or indentation.
+ */
+public class JavaDocFormatting extends AbstractJavadocCheck {
+
+    private static final String JAVA_DOC_RETURN = "javadoc return";
+    private static final String JAVA_DOC_PARAMETER = "javadoc parameter";
+    private static final String JAVA_DOC_THROW = "javadoc throw";
+
+    private static final String ERROR_THROW_WRONG_FORMAT = "Wrong format for javadoc throw statement";
```
Comment (conniey) [inline eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/JavaDocFormatting.java]: What is the right format?

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseHeaderDecoder.java [hunk 7]
```diff
@@ -75,7 +75,7 @@
      * @param headers the REST API returned headers
      * @return instance of header entity type created based on provided {@headers}, if header entity model does
      * not exists then return null
-     * @throws IOException
+     * @throws IOException If an I/O error occurs
```
Comment (conniey) [inline sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseHeaderDecoder.java]: Should we do a checkstyle ignore on these implementation classes?

File: sdk/keyvault/azure-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/SecretAsyncClient.java (hunk 5 above)
Comment (vhvb1989) [reply sdk/keyvault/azure-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/SecretAsyncClient.java]: will remove both. 👍

File: eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/JavaDocFormatting.java (hunk 6 above)
Comment (vhvb1989) [reply eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/JavaDocFormatting.java]: This is for an unrecognized format on a Javadoc throws..  something like 
`@throws` or `@throws 1 something`.

But, on second though, Javadoc parser would throw exception in that case, so, I will remove this better

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseHeaderDecoder.java (hunk 7 above)
Comment (vhvb1989) [reply sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseHeaderDecoder.java]: @JonathanGiles

Comment (JonathanGiles) [reply sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseHeaderDecoder.java]: Yes. We only care about public API. Things inside the implementation package are not public API and don't need to be checked for JavaDoc validity.

File: sdk/core/azure-core-management/src/test/java/com/azure/core/management/implementation/Value.java (hunk 1 above)
Comment (vhvb1989) [reply sdk/core/azure-core-management/src/test/java/com/azure/core/management/implementation/Value.java]: done

File: sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubAsyncProducerTest.java (hunk 2 above)
Comment (vhvb1989) [reply sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubAsyncProducerTest.java]: reverted and suppress added

File: sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/KeyClient.java (hunk 3 above)
Comment (vhvb1989) [reply sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/KeyClient.java]: ok

File: eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/JavaDocFormatting.java [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+/**
+ *  Description text should only have one space character after the parameter name or '@return' statement.
```
[the remark this answers is not in the document]
Comment (vhvb1989) [reply eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/JavaDocFormatting.java]: right. Fixed it

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseHeaderDecoder.java (hunk 7 above)
Comment (vhvb1989) [reply sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseHeaderDecoder.java]: Done.

File: sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/AsymmetricEncryptionAlgorithm.java (hunk 4 above)
Comment (vhvb1989) [reply sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/AsymmetricEncryptionAlgorithm.java]: @g2vinay if you share whatever text you want in there, I can make it part of this PR

File: eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/JavaDocFormatting.java [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+ *  Description text should only have one space character after the parameter name or {@code @return} statement.
+ *  Text should not start on a new line or have any additional spacing or indentation.
+ */
+public class JavaDocFormatting extends AbstractJavadocCheck {
+
+    private static final String JAVA_DOC_RETURN = "javadoc return";
+    private static final String JAVA_DOC_PARAMETER = "javadoc parameter";
+    private static final String JAVA_DOC_THROW = "javadoc throw";
+
+    private static final String ERROR_DESCRIPTION_ON_NEW_LINE = "Description for %s must be on same the same line";
```
Comment (conniey) [inline eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/JavaDocFormatting.java]: nit: Add a period to your sentences.

File: sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/AsymmetricEncryptionAlgorithm.java (hunk 4 above)
Comment (g2vinay) [reply sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/AsymmetricEncryptionAlgorithm.java]: > This exception text should be improved after this is merged (by @g2vinay)
This is not public API, 
I am just planning to keep the desrcription part of javadoc as a comment, will do it in a follow up clean up PR. @vhvb1989

**Pull request merged**